src/ILCompiler/src/Program.cs

```diff
@@ -67,61 +68,37 @@ private void InitializeDefaultOptions()
         // https://github.com/dotnet/corert/issues/568
         private void ParseCommandLine(string[] args)
         {
-            var parser = new CommandLineParser(args);
+            IReadOnlyList<string> explicitInputFiles = Array.Empty<string>();
+            IReadOnlyList<string> implicitInputFiles = Array.Empty<string>();
+            IReadOnlyList<string> referenceFiles = Array.Empty<string>();
 
-            string option;
-            while ((option = parser.GetOption()) != null)
+            AssemblyName name = typeof(Program).GetTypeInfo().Assembly.GetName();
+            ArgumentSyntax.Parse(args, syntax =>
             {
-                switch (option.ToLowerInvariant())
-                {
-                    case "?":
-                    case "help":
-                        _help = true;
-                        break;
-
-                    case "":
-                    case "in":
-                        parser.AppendExpandedPaths(_inputFilePaths, true);
-                        break;
-
-                    case "o":
-                    case "out":
-                        _options.OutputFilePath = parser.GetStringValue();
-                        break;
-
-                    case "dgmllog":
-                        _options.DgmlLog = parser.GetStringValue();
-                        break;
-
-                    case "fulllog":
-                        _options.FullLog = true;
-                        break;
-
-                    case "verbose":
-                        _options.Verbose = true;
-                        break;
-
-                    case "r":
-                    case "reference":
-                        parser.AppendExpandedPaths(_referenceFilePaths, false);
-                        break;
-
-                    case "cpp":
-                        _options.IsCppCodeGen = true;
-                        break;
-
-                    case "nolinenumbers":
-                        _options.NoLineNumbers = true;
-                        break;
-
-                    case "systemmodule":
-                        _options.SystemModuleName = parser.GetStringValue();
-                        break;
-
-                    default:
-                        throw new CommandLineException("Unrecognized option: " + parser.GetCurrentOption());
-                }
-            }
+                syntax.ApplicationName = name.Name + " " + name.Version.ToString();
+
+                syntax.HandleHelp = false;
```

Comment:
Why not let the help to be handled by Sysyem.CommandLine? (If there is a good reason - comment may be useful.)